lib/src/components/NxSearchTransferList/__tests__/NxSearchTransferList.test.tsx

```diff
@@ -5,160 +5,489 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
 
-import DataItem from '../../../util/DataItem';
-import { getShallowComponent } from '../../../__testutils__/enzymeUtils';
-import 'jest-enzyme';
-import NxSearchDropdown from '../../NxSearchDropdown/NxSearchDropdown';
-import NxTransferListHalf from '../../NxTransferListHalf/NxTransferListHalf';
+import React, { ComponentType } from 'react';
+import { includes } from 'ramda';
+import { within, render } from '@testing-library/react';
+import { rtlRender, rtlRenderElement, userEvent, runTimers } from '../../../__testutils__/rtlUtils';
 import NxSearchTransferList, { Props } from '../NxSearchTransferList';
 
 describe('NxSearchTransferList', function() {
-  const minimalProps = {
+  const minimalProps: Props = {
         searchText: '',
         onSearchTextChange: () => {},
         onSearch: () => {},
         loading: false,
         searchMatches: [],
         onSearchMatchSelect: () => {},
-        addedItems: [],
-        onRemove: () => {},
         addedItemsFilter: '',
-        onAddedItemsFilterChange: () => {}
+        onAddedItemsFilterChange: () => {},
+        addedItems: [],
+        onRemove: () => {}
       },
-      getShallow = getShallowComponent<Props<string | number>>(NxSearchTransferList, minimalProps);
+      quickRender = rtlRender(NxSearchTransferList as ComponentType<Props<string>>, minimalProps),
+      renderEl = rtlRenderElement(NxSearchTransferList as ComponentType<Props<string>>, minimalProps);
 
-  it('renders a div with the nx-search-transfer-list class', function() {
-    expect(getShallow()).toMatchSelector('div.nx-search-transfer-list');
-  });
+  it('adds specified classnames in addition to the defaults', function() {
+    const defaultEl = renderEl()!,
+        el = renderEl({ className: 'foo' })!;
 
-  it('adds additional specified div attrs', function() {
-    const component = getShallow({ id: 'foo', lang: 'en-US' });
+    expect(el).toHaveClass('foo');
 
-    expect(component).toHaveProp('id', 'foo');
-    expect(component).toHaveProp('lang', 'en-US');
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
-  it('adds additional specified class names', function() {
-    const component = getShallow({ className: 'foo' });
+  it('adds additional attributes to the top-level element', function() {
+    const el = renderEl({ id: 'foo', lang: 'en' });
 
-    expect(component).toHaveClassName('foo');
-    expect(component).toHaveClassName('nx-search-transfer-list');
+    expect(el).toHaveAttribute('id', 'foo');
+    expect(el).toHaveAttribute('lang', 'en');
```

Comment:
Since we're testing the "top-level" element, we should also test that `el` here is the top-level element?

Reply:
> Since we're testing the "top-level" element, we should also test that `el` here is the top-level element?

Sounds good! Tested here: [fc4cbc1](https://github.com/sonatype/sonatype-react-shared-components/pull/945/commits/fc4cbc14aa6082c2f3a01d511fc56153d7da99f4)